### PR TITLE
P01 Prototype

### PR DESCRIPTION
**Pull request by rpl-ffl**

### CODE REVIEW

File: release_testing/operational_tests/P01.jenkinsfile [hunk 1]
```diff
@@ -22,68 +22,315 @@ pipeline {
...
+	booleanParam(defaultValue: true, description: 'If checked, slope stage will be executed', name: 'enableSlope')
+	booleanParam(defaultValue: true, description: 'If checked, static stage will be executed', name: 'enableStatic')
+	booleanParam(defaultValue: false, description: 'If checked, A1 stage will be executed', name: 'enableA1')
+        booleanParam(defaultValue: false, description: 'If checked, A2 stage will be executed', name: 'enableA2')
+        booleanParam(defaultValue: true, description: 'If checked, B1 stage will be executed', name: 'enableB1')
+        booleanParam(defaultValue: true, description: 'If checked, B2 stage will be executed', name: 'enableB2')
+        booleanParam(defaultValue: false, description: 'If checked, B3 stage will be executed', name: 'enableB3')
+        booleanParam(defaultValue: false, description: 'If checked, B4 stage will be executed', name: 'enableB4')
+        booleanParam(defaultValue: false, description: 'If checked, B5 stage will be executed', name: 'enableB5')
+        booleanParam(defaultValue: false, description: 'If checked, C1 stage will be executed', name: 'enableC1')
```
Comment (Lubomir-Jahn) [inline release_testing/operational_tests/P01.jenkinsfile:35]: Please format the parameters in this fashion. 
```
booleanParam(
    name: 'enableDynamic',
    defaultValue: true,
    description: 'If checked, dynamic stage will be executed'
)
```

**Review by Lubomir-Jahn — CHANGES_REQUESTED** [review]

**Comment by Lubomir-Jahn** [comment]
Was this tested on jenkins already?

**Comment by rpl-ffl** [comment]
> Was this tested on jenkins already?

Tested here: https://scala.fantom.network/job/ReleaseTesting/job/Experimental/job/P01/20/

File: release_testing/operational_tests/P01.jenkinsfile (hunk 1 above)
Comment (rpl-ffl) [reply release_testing/operational_tests/P01.jenkinsfile]: Fixed here: https://github.com/Fantom-foundation/LaScala/pull/196/commits/a5374f64fddf1720f2b4dbae22c22850e0baf736

File: release_testing/operational_tests/P01.jenkinsfile [hunk 2]
```diff
@@ -22,68 +22,315 @@ pipeline {
...
+                    sh 'pwd'
+                    sh """../norma/build/norma run \
+                        -o /tmp/workspace/ReleaseTesting/OperationalTests/P01/runs \
+                        --label dynamic \
+                        ../norma/scenarios/demonet/dynamic.yml"""
+                }
+	    }
+            
+	    post {
+		always {
```
[the remark this answers is not in the document]
Comment (rpl-ffl) [reply release_testing/operational_tests/P01.jenkinsfile]: `post.always` -> `post.failure`
Only upload artifacts when run fails.

File: release_testing/operational_tests/P01.jenkinsfile [hunk 3]
```diff
@@ -22,68 +22,359 @@ pipeline {
             defaultValue: "main",
             description: 'Branch or commit hash for Norma'
         )
+	booleanParam(
+	    defaultValue: true, 
+	    description: 'If checked, dynamic stage will be executed', 
+	    name: 'enableDynamic'
+	)
```
Comment (Lubomir-Jahn) [inline release_testing/operational_tests/P01.jenkinsfile:29]: Convention suggested by Luis tells us that name of the parameter should be first. Could I ask you to do this additional change please? 🙏🏻 

Please do not forget all the other parameters.

**Review by Lubomir-Jahn — CHANGES_REQUESTED** [review]

File: release_testing/operational_tests/P01.jenkinsfile (hunk 3 above)
Comment (rpl-ffl) [reply release_testing/operational_tests/P01.jenkinsfile]: BooleanParam rearranged. Name first. https://github.com/Fantom-foundation/LaScala/pull/196/commits/43813dd44f8aed1831b4c1ea1899e914586d9eec

**Review by Lubomir-Jahn — APPROVED** [review]

**Pull request merged**